libsolidity/analysis/SyntaxChecker.cpp

```diff
@@ -299,19 +299,41 @@ bool SyntaxChecker::visit(ContractDefinition const& _contract)
 	return true;
 }
 
+void SyntaxChecker::endVisit(ContractDefinition const&)
+{
+	m_currentContractKind = {};
```

Comment:
```suggestion
	m_currentContractKind = std::nullopt;
```
A bit more explicit.